src/devices/Tcs3472x/Tcs3472x.cs

```diff
@@ -0,0 +1,227 @@
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.Drawing;
+using System.Threading;
+
+namespace Iot.Device.Tcs3472x
+{
+    public class Tcs3472xSensor : IDisposable
+    {
+        public const byte DefaultI2cAddress = 0x29;
+        private readonly I2cDevice _i2cDevice;
+        private byte _integrationTimeByte;
+        private double _integrationTime;
+        private bool _isLongTime;
+        private bool _autoDisposable;
+        private Gain _gain;
+
+        /// <summary>
+        /// Set/Get the time to wait for the sensor to read the data
+        /// Minimum time is 0.0024 s
+        /// Maximum time is 7.4 s
+        /// Be aware that it is not a linear function
+        /// </summary>
+        public double IntegrationTime
+        {
+            get { return _integrationTime; }
+            set
+            {
+                _integrationTime = value;
+                SetIntegrationTime(_integrationTime);
+            }
+        }
+
+        /// <summary>
+        /// Set/Get the gain
+        /// </summary>
+        public Gain Gain
+        {
+            get { return _gain; }
+            set
+            {
+                _gain = value;
+                _i2cDevice.Write(new byte[] { (byte)(Registers.COMMAND_BIT | Registers.CONTROL), (byte)_gain });
+            }
+        }
+
+        /// <summary>
+        /// Get the type of sensor
+        /// </summary>
+        public TCS3472Type ChipId { get; internal set; }
+
+        /// <summary>
+        /// Create a TCS4272x sensor
+        /// </summary>
+        /// <param name="i2cDevice">The I2C Device class</param>
+        /// <param name="integrationTime">The time to wait for sensor to read the data, minimum is 0.024 seconds, maximum in the constructor is 0.7 seconds</param>
+        /// <param name="gain">The gain when integrating the color measurement</param>
+        /// <param name="autoDisposable">true to dispose the I2C Device class at dispose</param>
+        public Tcs3472xSensor(I2cDevice i2cDevice, double integrationTime = 0.0024, Gain gain = Gain.GAIN_16X, bool autoDisposable = true)
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentException($"{nameof(i2cDevice)} can't be null");
+            // Maximum is 700 ms for the initialization. Value can be changed for a long one but not during this initialization phase            
+            _autoDisposable = autoDisposable;
+            _i2cDevice.WriteByte((byte)(Registers.COMMAND_BIT | Registers.ID));
+            ChipId = (TCS3472Type)_i2cDevice.ReadByte();
+            _isLongTime = false;
+            IntegrationTime = Math.Clamp(integrationTime, 0.0024, 0.7);
+            SetIntegrationTime(integrationTime);
+            Gain = gain;
+            PowerOn();
+        }
+
+        /// <summary>
+        /// Returns true is there are valid data
+        /// </summary>
+        /// <returns>Returns true is there are valid data</returns>
+        public bool IsValidData()
```

Comment:
I'd just make this property since we use that in most of the places for similar things